### ISSUE TITLE
Add Neighbor test

### ISSUE THREAD

**Issue by jessicaaustin** [ioos/ioos_qc#37]
This test compares sensor A to a co-located or sufficiently close sensor B measuring the same variable. 

*Note: There is also the Multi-variate test, that compares nearby sensors measuring *different* variables that should be related, like sea water temperature and salinity*

From the [QARTOD pH Manual](https://cdn.ioos.noaa.gov/media/2019/08/pHfinal_8_15_19b.pdf):

![Selection_891](https://user-images.githubusercontent.com/1159930/96903145-f02f6600-1441-11eb-8e52-7a9522889739.png)

And the [QARTOD T/S Manual](https://cdn.ioos.noaa.gov/media/2020/03/QARTOD_TS_Manual_Update2_200324_final.pdf):

![Selection_892](https://user-images.githubusercontent.com/1159930/96903194-03423600-1442-11eb-88a3-a86959965382.png)

In summary:

* The operator determines which other data stream(s) are sufficiently "close" to be used -- in other words, the test accepts data from sensor A (the sensor being tested) and sensor B (the neighbor) and assumes that it is appropriate to compare them
* Test inputs:
  * time and values for sensor A
  * time and values for sensor B
* Thresholds:
* Test algorithm:
  * See images above -- in short, if sensor A exceeds a rate of change threshold, and sensor B does not exceed it, then the data is suspect
* Results:
  * Returns PASS or SUSPECT - *never* FAIL

**Comment by jessicaaustin** [comment]
I've added a proposed test algorithm above that's based on the description in the QARTOD manual.

That said, I think there's also a need for a "neighbor test" that compares the *actual values* for the neighbor to the values for the sensor being tested? For example, here's a use case from @kbailey-noaa

> A NOS/CO-OPS rep just alerted me that 2 AOOS partner stations in Valdez (Valdez Duck Flats and Valdez Marine Terminal) were measuring air pressure ~ 10 mb lower than nearby stations.  
> Turns out the partner hadn't corrected the data to sea level, and it was subsequently fixed at the station ~10/6.  But I'm concerned because data were riding 10 mb low for however long, yet showing as 'passed' in AOOS QC tests: 
https://sensors.ioos.us/#metadata/100911/station/9/sensor 
> 
> I assume this is because those QC tests don't include neighbor checks.  I know there isn't a QARTOD manual for air pressure, and the neighbor test is usually only 'suggested' anyway.  But for baro this neighbor test is sometimes the only way to catch suspect data.  Is there any consideration for implementing neighbor checks?  
> Baro data tend to drift if the sensor isn't periodically calibrated, and it's concerning to think we'd have no way of catching that. For these 2 stations, the nearby CO-OPS valdez station should provide a useful comparison.
